src/client/components/Login.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import ReactDOM from "react-dom/client";
+import "../styles/Login.css";
+
+import { BrowserRouter as Router, Route, Link } from "react-router-dom";
+import Owner from "./Owner";
+import Createacc from "./Createacc";
+
+interface ErrorMessage {
+  name: string;
+  message: string;
+}
+
+interface FormValue {
+  name: string;
+  pass: string;
+}
+
+function Login() {
+  //seed data
+  const database = [
+    {
+      username: "user1",
+      password: "pass1",
+    },
+    {
+      username: "user2",
+      password: "pass2",
+    },
+  ];
+
+  // React States
+  const [errorMessages, setErrorMessages] = useState<ErrorMessage>({
+    name: "",
+    message: "",
+  });
+  const [loginSuccess, setLoginSuccess] = useState<boolean>(false);
+  const [formValue, setFormValue] = useState<FormValue>({
+    name: "",
+    pass: "",
+  });
+
+  // User Login info
+  const loginErrors = {
+    uname: "invalid username",
+    pass: "invalid password",
+  };
+
+  //Update Input
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormValue((prevState) => {
+      return {
+        ...prevState,
+        [name]: value,
+      };
+    });
+  };
+
+  //Handle Submit
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    // Check if username is correct
+    const userData = database.find((user) => user.username === formValue.name);
+
+    // Check if user pass is corect
+    if (userData) {
+      if (userData.password !== formValue.pass) {
+        // Invalid password
+        setErrorMessages({ name: "pass", message: loginErrors.pass });
+      } else {
+        setLoginSuccess(true);
+      }
+    } else {
+      // Username not found
+      setErrorMessages({ name: "name", message: loginErrors.uname });
+    }
+  };
+
+  // Generate error message
+  const renderErrorMessage = (name: string) => {
+    if (name === errorMessages.name) {
+      return <div className="error">{errorMessages.message}</div>;
+    } else {
+      return undefined;
+    }
```

Comment:
I think this `else` is unnecessary. Could you try and test it?
Also add the return type for this function.

Reply:
You are correct!